android/src/ti/intercom/TitaniumIntercomModule.java

```diff
@@ -54,7 +56,13 @@ public static void onAppCreate(TiApplication app) {
 		apiKey = app.getAppProperties().getString("intercom-android-api-key", null);
 		appId = app.getAppProperties().getString("intercom-app-id", null);
 
-		initialize();
+		// Defer the initialization on Android 12 and 13 to avoid ANRs as suggested by the Intercom Support team.
+		// It would further require that the Intercom.initialize() is called before any other Intercom API is used.
+		if (Build.VERSION.SDK_INT >= 31 && Build.VERSION.SDK_INT <= 33) {
```

Comment:
Is there a reason why only those 2 versions are affected?

Reply:
Couldn't find any solid reasons, our Firebase reports provides this data. So we want to keep the changes around the most impacted versions only as this work-around from Intercom Support is more of a mitigation step.

However, I did a decompile on Intercom's Android SDK and found that most of their primary methods are `synchronized`, indicating potential thread-locking at multiple places.